pennylane/measurements.py

```diff
@@ -587,3 +593,115 @@ def circuit():
+            measurement outcome was obtained
+        one_case (float): the measurement outcome in case the second
+            measurement outcome was obtained
+    """
+
+    __slots__ = ("_depends_on", "_zero_case", "_one_case", "_control_value")
+
+    def __init__(
+        self,
+        measurement_id: str,
+        zero_case: float = 0,
+        one_case: float = 1,
+    ):
+        self._depends_on = measurement_id
+        self._zero_case = zero_case
+        self._one_case = one_case
+        self._control_value = one_case  # By default, control on the one case
+
+    @property
+    def branches(self):
+        """A dictionary representing all the possible outcomes of the MeasurementValue."""
+        branch_dict = {}
+        branch_dict[(0,)] = self._zero_case
+        branch_dict[(1,)] = self._one_case
+        return branch_dict
+
+    def __eq__(self, control_value):
+        """Allow asserting measurement values."""
+        measurement_outcomes = {self._zero_case, self._one_case}
+        if control_value not in measurement_outcomes:
+            raise MeasurementValueError(
+                f"Unknown measurement value asserted; the set of possible measurement outcomes is: {measurement_outcomes}."
+            )
+
+        self._control_value = control_value
+        return self
+
+    @property
+    def control_value(self):
+        """The control value to consider for the measurement outcome."""
+        return self._control_value
+
+    @property
+    def measurements(self):
+        """List of all measurements this MeasurementValue depends on."""
+        return [self._depends_on]
+
+
+def measure(wires):
+    """Perform a mid-circuit measurement in the computational basis on the
+    supplied qubit.
+
+    Measurement outcomes can be obtained and used to conditionally apply
+    operations.
+
+    If a device doesn't support mid-circuit measurements natively, then the
+    QNode will apply the :func:`defer_measurements` transform.
+
+    **Example:**
+
+    .. code-block:: python3
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def func(x, y):
+            qml.RY(x, wires=1)
+            qml.CNOT(wires=[0, 1])
+            m_0 = qml.measure(1)
+
+            qml.cond(m_0, qml.RY)(y, wires=0)
+            return qml.probs(wires=[0])
+
+    Executing this QNode:
+
+    >>> pars = np.array([0.643, 0.246], requires_grad=True)
+    >>> func(*pars)
+    tensor([0.99849698, 0.00150302], requires_grad=True)
+
+    Args:
+        wires (.Wires): The wires the measurement process applies to.
+
+    Raises:
+        QuantumFunctionError: if multiple wires were specified
+    """
+    wire = qml.wires.Wires(wires)
+    if len(wire) > 1:
+        raise qml.QuantumFunctionError(
+            "Only a single qubit can be measured in the middle of the circuit"
+        )
+
+    # Create a UUID and a map between MP and MV to support serialization
+    measurement_id = str(uuid.uuid4())[:8]
```

Comment:
Do we keep the first 8 characters only for readability? How confident are we in the uniqueness of the first 8?

Reply:
@puzzleshark any thoughts here?

Reply:
Ya, it's for readability. I think in our case it's probably fine, and can always be changed